pkg/controller/endpointslice/topologycache/utils.go

```diff
@@ -185,6 +189,9 @@ func getMost(zones map[string]float64) (string, float64) {
 func getHintsByZone(slice *discovery.EndpointSlice, allocatedHintsByZone EndpointZoneInfo, allocations map[string]Allocation) map[string]int {
 	hintsByZone := map[string]int{}
 	for _, endpoint := range slice.Endpoints {
+		if endpoint.Conditions.Ready == nil || !*endpoint.Conditions.Ready {
```

Comment:
(maybe a helper function for `if endpoint.Conditions.Ready == nil || !*endpoint.Conditions.Ready`?)

Reply:
Updated, thanks!